client/src/Context/AuthContext.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, createContext, useEffect, ReactNode } from "react";
+import { useNavigate } from "react-router-dom";
+import { getToken } from "../utils/getToken";
+
+interface AuthContextType {
+  newUser: SignUp;
+  selectedFile: File | string;
+  submitImage: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  signUp: () => void;
+  logIn: () => void;
+  setSelectedFile: (selectedFile: File | string) => void;
+  setNewUser: (newUser: SignUp) => void;
+  user: boolean;
+  setUser: (user: boolean) => void;
+  loginUser: Login;
+  setLoginUser: (loginUser: Login) => void;
+  logOut: () => void;
+}
+interface Props {
+  children: ReactNode;
+}
+
+//TODO decide if array should include object id or something else, then change 'any';
+interface LoginResult {
+  accessToken: string;
+  isAuthenticated: boolean;
+  message: string;
+  user: {
+    email: string;
+    firstName: string;
+    image: string;
+    lastName: string;
+    starredCoworking: Array<any>;
+    starredMentorship: Array<any>;
+    starredShadowing: Array<any>;
+    username: string;
+    _id: number;
+  };
+}
+
+//TODO add image in respone?
```

Comment:
Yes, please add it, maybe with a question mark as an optional? 

**Current status:** 
-> mongoose user model has this field, but it's not marked as a required one
-> if an image is uploaded, the link should be stored in mongo
-> and passed on to the response here; 
-> mongoose seems to not be adding the field if no value is given;
-> hence all test users without an image do not have this key

What I don't get is: _why is mongoose not adding the field, just as it is adding every other one, eg. starredCoworking etc. (also not marked required in the db)?_